Source/SwiftLintFramework/Rules/ImplicitGetterRule.swift

```diff
@@ -77,34 +197,43 @@ public struct ImplicitGetterRule: ConfigurationProviderRule {
             return token
         }
 
-        let violatingTokens = getTokens.filter { token -> Bool in
+        let violatingLocations = getTokens.compactMap { token -> (Int, SwiftDeclarationKind?)? in
             // the last element is the deepest structure
-            guard let dict = variableDeclarations(forByteOffset: token.offset, structure: file.structure).last else {
-                return false
+            guard let dict = declarations(forByteOffset: token.offset, structure: file.structure).last else {
+                return nil
             }
 
             // If there's a setter, `get` is allowed
-            return dict.setterAccessibility == nil
+            guard dict.setterAccessibility == nil else {
+                return nil
+            }
+
+            let kind = dict.kind.flatMap(SwiftDeclarationKind.init(rawValue:))
+            return (token.offset, kind)
         }
 
-        return violatingTokens.map { token in
-            // Violation found!
-            let location = Location(file: file, byteOffset: token.offset)
+        return violatingLocations.map { offset, kind in
+            let reason = kind.map { kind -> String in
+                let kind = kind == .functionSubscript ? "subscripts" : "properties"
```

Comment:
Can you rename this to `let kindString` or something? So many different `kind` identifiers in these 2 lines.

Reply:
Hah, nice catch 😅